src/app/components/site/Homepage.tsx

```diff
@@ -56,9 +61,11 @@ export const Homepage = () => {
                                     <ShowMeButtons className={"pt-xl-2"}/>
                                     {/*<img id="homepageHeroImg" className="img-fluid" alt="Three Computer Science students studying with two laptops, one with code on the screen" src="/assets/ics_hero.svg" />*/}
                                 </Col>
-                                <Col data-testid={"featured-news-item"} md="12" lg="7" className="d-none d-lg-block">
+                                {user?.role === "TEACHER" && promo ? (<Col data-testid={"promo-tile"} md="12" lg="7" className="mt-4 d-block">
```

Comment:
This is hard to read and a bit convoluted. A cleaner alternative would be to move this into its own small component (it can live in this file). The component could have the same shape: it would be a `Col` that has another component inside (either TeacherPromoItem or FeaturedNewsItem). So the base shape looks like this:
```jsx
<Col data-testid={"AAAA"} md="12" lg="7" className="BBBB d-lg-block">
    <CCCC item={DDDD} />
</Col>
```
Inside the component you decide what AAAA, BBBB, CCCC and DDDD look like depending on your conditions (which are whether the user is a teacher and there is a promo). But AAAA, BBBB, CCCC and DDDD are all variables.

By doing it like this, you extract this functionality into its own component, which makes this part of the code easier to read, and the component itself encapsulates all the logic that it needs, which makes it easier to understand. In addition, the shape is a lot more clear because rather than repeating it twice (like it's done currently), you just write it once and control how it looks like with variables.

Reply:
Refactored this one into `PromoOrFeaturedNews`, now living inside `Dashboard` component to make `Homepage` more readable and less cluttered.